hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -412,6 +412,28 @@ private void validateTableIsIcebergTableOrView(
     }
   }
 
+  @Override
+  public boolean tableExists(TableIdentifier identifier) {
+    if (!isValidIdentifier(identifier)) {
+      return false;
+    }
+
+    String database = identifier.namespace().level(0);
+    try {
+      Table table = clients.run(client -> client.getTable(database, identifier.name()));
+      HiveOperationsBase.validateTableIsIceberg(table, fullTableName(name, identifier));
+      return true;
+    } catch (NoSuchTableException | NoSuchObjectException e) {
+      return false;
```

Comment:
what happens if a hive table with the same name exists in HMS? I think `tableExists` will return false, which is confusing.

Reply:
I think the HMS call is getTable (line 423), not tableExists as I also was confused in https://github.com/apache/iceberg/pull/11597/files#r1853126846, so it should work iiuc

Reply:
+1 to keep the behavior the same. Lets add a test to explicitly call out this behavior

Reply:
I added test to cover this and also there's existing test `testHiveTableAndIcebergTableWithSameName` to cover as well